bench/textx/json.py

```diff
@@ -0,0 +1,50 @@
+from textx import metamodel_from_file
+from os.path import join, dirname
+import collections
+from bench import helpers
+
+
+def compile():
+    class Member:
+        def __init__(self, key, value, parent=None):
+            self.pair = (key, value)
+
+        def __repr__(self):
+            return str(self.pair)
+
+        def __iter__(self):
+            return iter(self.pair)
+
+    class Object(collections.UserDict):
+        def __init__(self, members, parent=None):
+            self.data = {key: value for key, value in members}
+
+    class Array(collections.UserList):
+        def __init__(self, values, parent=None):
+            self.data = list(values)
+
+    json_mm = metamodel_from_file(
+        join(dirname(__file__), "json.tx"),
+        debug=False,
+        classes=[Member, Object, Array],
+        memoization=True,
```

Comment:
For JSON, memoization actually hurts performance because there's basically no backtracking needed.